src/enum_net.cpp

```diff
@@ -697,12 +846,39 @@ int _System __libsocket_sysctl(int* mib, u_int namelen, void *oldp, size_t *oldl
 				r.friendly_name[sizeof(r.friendly_name) - 1] = '\0';
 				wcstombs(r.description, adapter->Description, sizeof(r.description));
 				r.description[sizeof(r.description) - 1] = '\0';
+				r.state
+					= (adapter->OperStatus == IfOperStatusUp) ? if_state::up
+					: (adapter->OperStatus == IfOperStatusDown) ? if_state::down
+					: (adapter->OperStatus == IfOperStatusTesting) ? if_state::testing
+					: (adapter->OperStatus == IfOperStatusUnknown) ? if_state::unknown
+					: (adapter->OperStatus == IfOperStatusDormant) ? if_state::dormant
+					: (adapter->OperStatus == IfOperStatusNotPresent) ? if_state::notpresent
+					: (adapter->OperStatus == IfOperStatusLowerLayerDown) ? if_state::lowerlayerdown
+					: if_state::unknown;
+
+				r.flags = r.state != if_state::down ? if_flags::up : interface_flags{};
+				if (adapter->IfType == IF_TYPE_SOFTWARE_LOOPBACK)
+					r.flags |= if_flags::loopback;
+				if (adapter->IfType == IF_TYPE_PPP)
+					r.flags |= if_flags::pointopoint;
+				if (!(adapter->Flags & IP_ADAPTER_NO_MULTICAST))
```

Comment:
This flag is actually the opposite meaning of the flag on other platform.

Reply:
oh, you mean on windows this means whether it's a multicast *recipient* whether on other platforms the multicast flag means it's multicast *capable*?

Reply:
it doesn't sound too different though. form msdn: "The adapter is not a multicast recipient."
versus the linux `IFF_MULTICAST` flag: "Supports multicast"

Reply:
Oops my mistake I missed the exclamation point in front.